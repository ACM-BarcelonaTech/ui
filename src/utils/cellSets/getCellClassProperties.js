import _ from 'lodash';

/**
 * Returns the properties of the first of the cellSets that contains the cellId
 * (by the order specified in hierarchy children)
 * @param {int} cellId The id of the cell we are searching for.
 * @param {string} cellSetClassKey The cellSetClass we are searcing inside of
 *  (e.g.: sample, louvain, scratchpad, etc..).
 * @param {array} hierarchy CellSets hierarchy.
 * @param {object} properties CellSets properties.
 */

const getCellClassProperties = (cellId, cellSetClassKey, cellSets) => {
  const { properties, hierarchy } = cellSets;
  console.log('cell colour ', cellSetClassKey, cellSets);

  let childrenCellSets = hierarchy.filter(({ key }) => cellSetClassKey.includes(key))
    .reduce((prev, curr) => prev.children.concat(curr.children)).children;
  console.log('yaaaaay', childrenCellSets);
  const cellSetsContainingCell = [];
  childrenCellSets = _.isArray(childrenCellSets) ? childrenCellSets : [childrenCellSets];

  childrenCellSets.forEach(({ key }) => {
    if (properties[key].cellIds.has(cellId)) {
      cellSetsContainingCell.push(key);
    }
  });
  // console.log('CONTAINING CELLSET')
  const clusterProperties = cellSetsContainingCell.map((clusterKey) => {
    const rootClusterName = hierarchy.filter(({ children }) => children.filter((child) => child.key === clusterKey).length > 0)[0].key;
    console.log(properties[clusterKey], 'hi :))');
    return { ...properties[clusterKey], rootClusterName: _.capitalize(rootClusterName) };
  });
  console.log('childrenCellSets2', clusterProperties);

  return clusterProperties;
};

export default getCellClassProperties;
