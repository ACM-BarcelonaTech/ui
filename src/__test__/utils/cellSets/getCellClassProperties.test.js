import getCellClassProperties from 'utils/cellSets/getCellClassProperties';
import mockCellSets from 'utils/tests/mockStores/cellSets';

describe('get cell class properties tests', () => {
  const { cellSets } = mockCellSets();
  it('returns information for a passed cellId', () => {
    // doing it as a string to make sure it still works (sometimes a string is passed from vitessce)
    const cellId = '3';
    const cellSetClassKeys = ['test'];
    const result = getCellClassProperties(cellId, cellSetClassKeys, cellSets);
    expect(result).toMatchSnapshot();
  });
});
