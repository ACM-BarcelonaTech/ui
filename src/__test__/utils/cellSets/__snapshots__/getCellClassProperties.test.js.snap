// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get cell class properties tests returns information for a passed cellId 1`] = `
Array [
  Object {
    "cellIds": Set {
      1,
      2,
      3,
    },
    "clusterName": "Test : Test-1",
    "name": "Test-1",
  },
]
`;
